core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMNode.kt

```diff
@@ -61,4 +91,8 @@ data class STDCMNode(
             waypointIndex
         )
     }
+
+    private fun areTimesDifferent(time1: Double, time2: Double): Boolean {
+        return abs(time1 - time2) > 1e-3
+    }
```

Comment:
There's already `TrainPhysicsIntegrator.areTimesEqual`. Though the epsilon is different there (1e-2).

Reply:
I've actually been thinking about it during the afternoon, after @Khoyo 's comment. Since we want to fill the priority queue in the correct order, it made sense to me to compare the times strictly and use 1e-3 (i.e. precision to the ms). But that level of precision seems somewhat over kill (especially since we now have a lot of levels of conditional priority). I'm down to just use the standard `areTimesEqual` with a precision of 1e-2. Are you ok with that ?